sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/ChangeFeedProcessorState.java

```diff
@@ -0,0 +1,193 @@
+
+    /**
+     * Sets the token of the lease item representing the persistent state of a change feed processor worker.
+     * <p>
+     * A lease token is a unique identifier representing a specific scope that a CFP worker will operate on.
+     *
+     * @param leaseToken a unique string representing a specific scope that a CFP worker will operate on.
+     * @return the current ChangeFeedProcessorState instance.
+     */
+    public ChangeFeedProcessorState setLeaseToken(String leaseToken) {
+        this.leaseToken = leaseToken;
+        return this;
+    }
+
+    /**
+     * Gets the name of the host which operates on the lease item.
+     * <p>
+     * When using multiple CFP instances distributing the work for a given feed container, each host must have a unique name.
+     *
+     * @return the host name that has ownership of this lease item or null if no host is currently operating on this lease.
+     */
+    public String getHostName() {
+        return this.hostName;
+    }
+
+    /**
+     * Sets the name of the host which operates on the lease item.
+     * <p>
+     * When using multiple CFP instances distributing the work for a given feed container, each host must have a unique name.
+     *
+     * @param hostName the host name that has ownership of this lease item.
+     * @return the current ChangeFeedProcessorState instance.
+     */
+    public ChangeFeedProcessorState setHostName(String hostName) {
+        this.hostName = hostName;
+        return this;
+    }
+
+    /**
+     * Gets the time when the lease item was last updated.
+     *
+     * @return time when the lease item was last updated.
+     */
+    public Instant getLastUpdatedTime() {
+        return this.lastUpdatedTime;
+    }
+
+    /**
+     * Sets time when the lease item was last updated.
+     *
+     * @param lastUpdatedTime a unique string.
+     * @return the current ChangeFeedProcessorState instance.
+     */
+    public ChangeFeedProcessorState setLastUpdatedTime(Instant lastUpdatedTime) {
+        this.lastUpdatedTime = lastUpdatedTime;
+        return this;
+    }
+
+    /**
+     * Gets a marker representing the last item that was processed.
+     *
+     * @return the marker representing the last item that was processed.
+     */
+    public String getContinuationToken() {
+        return this.continuationToken;
+    }
+
+    /**
+     * Sets a marker representing the last item that was processed.
+     *
+     * @param continuationToken the marker representing the last item that was processed.
+     * @return the current ChangeFeedProcessorState instance.
+     */
+    public ChangeFeedProcessorState setContinuationToken(String continuationToken) {
+        this.continuationToken = continuationToken;
+        return this;
+    }
+
+    /**
+     * Gets the system time for the last item that was processed.
+     *
+     * @return the system time for the last item that was processed.
+     */
+    public Instant getContinuationTokenTimestamp() {
```

Comment:
_ts is implementation detail. And it might get impacted by the clock skews on service side. 
Its better for any application to not take dependency on this concept/value.

Reply:
I will take it out; it's not needed for the customer scenario we want to enable as preview.